File: app/src/main/java/com/example/rxjavademo/SplashScreenActivity.java
package com.example.rxjavademo;

import android.annotation.SuppressLint;
import android.util.Log;
import android.widget.Button;
import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import io.reactivex.Flowable;
import io.reactivex.Observable;

public class SplashScreenActivity extends AppCompatActivity {

    @SuppressLint("CheckResult")
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash_screen);
        Button btn = findViewById(R.id.button);

        int startValue = 1;
        int incrementValue = 1;
//        Flowable.fromCallable(() -> doSomeThing(5))
//                .map(integer -> integer +10)
//                .subscribe(value -> Log.d("fromCallable", ""+value));

        Flowable<Integer> flowable = Flowable.generate(() -> startValue, (s, emitter) -> {
            int nextValue = s + incrementValue;
            if (nextValue == 1000) {
                emitter.onComplete();
            } else {
                emitter.onNext(nextValue);
            }
            return nextValue;
        }, i -> Log.d("sss", ""+ i.toString()));

        flowable.subscribe(value -> Log.d("generate", "" + value));

        btn.setOnClickListener(view -> Observable.generate(emitter -> {
            emitter.onNext(1);
            emitter.onComplete();
        })
                .subscribe(value -> Log.d("ButtonClicked", "" + value)));
    }

    private int doSomeThing(int i) {
        return i+5;
    }
}

File: app/src/main/java/com/example/rxkotlin/SplashScreenActivity.kt
package com.example.rxkotlin

import android.annotation.SuppressLint
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import io.reactivex.Single
import java.util.concurrent.TimeUnit

class SplashScreenActivity : AppCompatActivity() {

    @SuppressLint("CheckResult")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash_screen)

        Single.timer(5, TimeUnit.SECONDS).subscribe { _ -> changeActivity() }
    }

    private fun changeActivity() {
        val intent = Intent(this, MainActivity::class.java)
        startActivity(intent)
    }
}
